### PR TITLE
Improve tests for JavaScript

### PR DESCRIPTION
**Pull request by ykzts**
- Upgrade dependencies
    - chai (3.5.0 -> 4.0.1)
    - chai-enzyme (0.6.1 -> 0.7.1)
    - sinon (2.2.0 -> 2.3.2)
- Change extensions from `.jsx` to `.js`
- Don't assign `React` to `global`
- Check code format using ESLint

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -116,8 +116,8 @@
   "devDependencies": {
     "@kadira/storybook": "^2.35.3",
     "babel-eslint": "^7.2.3",
-    "chai": "^3.5.0",
-    "chai-enzyme": "^0.6.1",
+    "chai": "^4.0.1",
```
Comment (ykzts) [inline package.json]: https://github.com/chaijs/chai/releases/tag/4.0.0

> ### Breaking Changes
>
> We no longer support Node v0.10 and v0.12 (since their LTS has ended) _(PRs: [#816](https://github.com/chaijs/chai/pull/816), [#901](https://github.com/chaijs/chai/pull/901))_
>
> ...

Other changes are unrelated changes to Mastodon.

**Review by abcang — APPROVED** [review]
